### PR TITLE
ISPN-13229 LocalPublisherManagerImpl doesn't clean up changeListener …

### PR DESCRIPTION
**Pull request by wburns**
…for publish methods

https://issues.redhat.com/browse/ISPN-13229

### CODE REVIEW

File: core/src/main/java/org/infinispan/reactive/publisher/impl/LocalPublisherManagerImpl.java [hunk 1]
```diff
@@ -241,18 +241,18 @@ private SegmentAwarePublisherImpl(IntSet segments, CacheSet<I> set,
...
+         Flowable<R> resultPublisher;
          switch (deliveryGuarantee) {
             case AT_MOST_ONCE:
-                segmentPublishers = Flowable.fromStream(segments.intStream().mapToObj(segment -> {
-                   Publisher<I> publisher = set.localPublisher(segment);
-                   if (predicate != null) {
-                      publisher = Flowable.fromPublisher(publisher)
-                            .filter(predicate);
-                   }
+               resultPublisher = Flowable.fromStream(segments.intStream().mapToObj(segment -> {
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/reactive/publisher/impl/LocalPublisherManagerImpl.java:247]: Do we really need to use streams here?
Wouldn't it be better to use `Flowable.fromIterable(segments).concatMap(segment -> ...)`?
(Ideally on separate lines so the `.` on line 286 can be aligned to the previous `.` as well.)

Comment (wburns) [reply core/src/main/java/org/infinispan/reactive/publisher/impl/LocalPublisherManagerImpl.java:247]: Sure, I can add that change as part of this PR.

**Review by danberindei — APPROVED** [review]

**Comment by danberindei** [comment]
@wburns there's a failure in CI that only appears when you click the "Test Result" link

https://ci.infinispan.org/job/Infinispan/job/PR-9481/2/testReport/org.infinispan.distribution.rehash/StateTransferOverwriteTest/createBeforeMethod/

I don't see any connection to your changes, and the failure also appears in other PR builds (e.g. https://ci.infinispan.org/job/Infinispan/job/PR-9462/1/), so I'm merging the PR and we can handle the NPE later.

**Comment by danberindei** [comment]
I fixed the `StateTransferOverwriteTest` failure in #9483

**Pull request merged**